tICA/tICAPipeline.sh

```diff
@@ -0,0 +1,680 @@
+    export HCPPIPEDIR="$(dirname -- "$0")/.."
+fi
+
+source "$HCPPIPEDIR/global/scripts/newopts.shlib" "$@"
+source "$HCPPIPEDIR/global/scripts/debug.shlib" "$@"
+source "$HCPPIPEDIR/global/scripts/tempfiles.shlib" "$@"
+source "$HCPPIPEDIR/global/scripts/parallel.shlib" "$@"
+g_matlab_default_mode=1
+#add steps to this array and in the switch cases below
+pipelineSteps=(MIGP GroupSICA indProjSICA ConcatGroupSICA ComputeGroupTICA indProjTICA ComputeTICAFeatures ClassifyTICA CleanData)
+defaultStart="${pipelineSteps[0]}"
+defaultStopAfter="${pipelineSteps[${#pipelineSteps[@]} - 1]}"
+stepsText="$(IFS=$'\n'; echo "${pipelineSteps[*]}")"
+
+#description to use in usage - syntax of parameters is now explained automatically
+opts_SetScriptDescription "implements temporal ICA decomposition and cleanup"
+
+#mandatory (mrfix name must be specified if applicable, so including it here despite being mechanically optional)
+#general inputs
+opts_AddMandatory '--study-folder' 'StudyFolder' 'path' "folder that contains all subjects"
+opts_AddMandatory '--subject-list' 'SubjlistRaw' '100206@100307...' "list of subject IDs separated by @s"
+opts_AddMandatory '--fmri-names' 'fMRINames' 'rfMRI_REST1_LR@rfMRI_REST1_RL...' "list of fmri run names separated by @s" #Needs to be the single fMRI run names only (for DVARS and GS code) for MR+FIX, is also the SR+FIX input names
+opts_AddOptional '--mrfix-concat-name' 'MRFixConcatName' 'rfMRI_REST' "if multi-run FIX was used, you must specify the concat name with this option"
+opts_AddMandatory '--output-fmri-name' 'OutputfMRIName' 'rfMRI_REST' "name to use for tICA pipeline outputs"
+opts_AddMandatory '--proc-string' 'fMRIProcSTRING' 'string' "file name component representing the preprocessing already done, e.g. '_Atlas_MSMAll_hp0_clean'"
+opts_AddMandatory '--fix-high-pass' 'HighPass' 'integer' 'the high pass value that was used when running FIX' '--melodic-high-pass'
+opts_AddMandatory '--out-group-name' 'GroupAverageName' 'string' 'name to use for the group output folder'
+opts_AddMandatory '--fmri-resolution' 'fMRIResolution' 'string' "resolution of data, like '2' or '1.60' "
+#TSC: doesn't default to MSMAll because we don't have that default string in the MSMAll pipeline
+opts_AddMandatory '--surf-reg-name' 'RegName' 'MSMAll' "the registration string corresponding to the input files"
+#sICA
+opts_AddConfigMandatory '--num-wishart' 'numWisharts' 'numWisharts' 'integer' "how many wisharts to use in icaDim" #FIXME - We will need to think about how to help users set this.  Ideally it is established by running a null model, but that is timeconsuming. Valid values for humans have been WF5 or WF6.
+#sICA individual projection
+opts_AddConfigMandatory '--low-res' 'LowResMesh' 'LowResMesh' 'meshnum' "mesh resolution, like '32' for 32k_fs_LR"
+opts_AddMandatory '--subject-expected-timepoints' 'subjectExpectedTimepoints' 'string' "output spectra size for sICA individual projection, RunsXNumTimePoints, like '4800'"
```

Comment:
So, what if a given subject has less than the "expected" number?  Or if processing a single subject, must it be the actual number of TPs?

Reply:
Some parts of the code pad subjects with zeros to match, others just use it to normalize things...I don't know the particulars of what we tested, we may have included a partial-data subject...